### PR TITLE
Enhance memory map documentation

### PR DESCRIPTION
**Pull request by atharvalele**
- Add details on types of memory and memory allocation
- Add details on memory allocation libraries used

### CODE REVIEW

File: source/developer-guides/memory-map.rst [hunk 1]
```diff
@@ -47,13 +54,96 @@ Permanent Memory Map
...
+
+* ``MemoryAllocationLib`` provides services for allocating two different types of memory -
+
+  * Bootloader permanent memory
+
+    * Note that permanent memory does not mean that it is persistent across reboots
+
+  * Temporary memory
+
+* Bootloader permanent memory is used by the bootloader to store device tables, S3 Data,
```
Comment (rprangar) [inline source/developer-guides/memory-map.rst:91]: Please consolidate Permanent Memory and Temp Memorary details

File: source/developer-guides/memory-map.rst [hunk 2]
```diff
@@ -47,13 +54,96 @@ Permanent Memory Map
...
+  SBL container headers, etc.
+* Temporary memory requested by a stage is expected to be used by that stage itself.
+
+* ``MemoryAllocationLib`` allocates memory from the Loader Reserved Memory region
```
Comment (rprangar) [inline source/developer-guides/memory-map.rst:95]: Maybe start the section with this line (move this line to 83)?

**Comment by atharvalele** [comment]
Thanks for the review. Updated the PR @rprangar

**Review by rprangar — COMMENTED** [review]
LGTM

**Review by rprangar — APPROVED** [review]
LGTM

**Pull request merged**